py_gtfs_rt_ingestion/batch_files.py

```diff
@@ -0,0 +1,76 @@
+#!/usr/bin/env python
+
+import argparse
+import boto3
+import sys
+
+from collections.abc import Iterable
+
+from py_gtfs_rt_ingestion import batch_files
+
+DESCRIPTION = "Generate batches of json files that should be processesd"
+
+def parseArgs(args) -> dict:
+    parser = argparse.ArgumentParser(description=DESCRIPTION)
+    parser.add_argument(
+        '--input',
+        dest='input_file',
+        type=str,
+        help='text file of AWS S3 ls call')
+
+    parser.add_argument(
+        '--s3-prefix',
+        dest='s3_prefix',
+        type=str,
+        help='prefix to files in the mbta-gtfs-s3 bucket')
+
+    parser.add_argument(
+        '--threshold',
+        dest='filesize_threshold',
+        type=int,
+        default=100000,
+        help='filesize threshold for batch sizes')
+
+    parsed_args = parser.parse_args(args)
+
+    return parsed_args
+
+def file_list_from_file(input_file: str) -> Iterable[(str, int)]:
+    """
+    take a file that is the output of `aws s3 ls <dir>` and yield out filename,
+    filesize tuples.
+    """
+    for file_info in open(input_file):
+        (date, time, size, filename) = file_info.split()
+        yield (filename, sie)
+
+def file_list_from_s3(file_prefix) -> Iterable[(str, int)]:
+    """
+    look in the mbta-gtfs-s3 bucket for files begining with this prefix and
+    yield out filename, filesize tuples.
+    """
+    s3 = boto3.resource('s3')
+    bucket = s3.Bucket('mbta-gtfs-s3')
+
+    for bucket_object in bucket.objects.filter(Prefix=file_prefix):
+        yield (bucket_object.key, bucket_object.size)
```

Comment:
generator functions that produce tuples of filenames and filesizes that will be batched.

Reply:
It might not matter for how we're going to be using this, but I think all standard `filter` methods are limited to returning 1000 items. 

[S3.Bucket.objects](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/s3.html?highlight=bucket#S3.Bucket.objects)
 > MaxKeys (integer) -- Sets the maximum number of keys returned in the response. By default the action returns up to 1,000 key names. The response might contain fewer keys but will never contain more.

`Paginators` are recomended for any call that could return more than 1000 results: https://boto3.amazonaws.com/v1/documentation/api/latest/guide/paginators.html